### DISCUSSION TITLE
Is possible to train 2 textcat components into the same pipeline

### DISCUSSION THREAD

**Discussion by info2000** [explosion/spaCy#10692] [Help: Coding & Implementations] [answered]
I need to have 2 textcat on the same pipeline, _cats and _catsB
is possible to train both at the same time? or should I train one first and next add the trained model as component's source ?
if is possible, what is the doc.key to pass the _catsB component annotations?

Thanks

**Comment by thomashacker** [comment, marked as answer]
Hello,
You can't train at the same time because there's only one place to represent the training data in the doc. As you mentioned, you'll need to train one component first and source it to the other pipeline.
